src/components/AllSpecies/AllSpecies.tsx

```diff
@@ -1,41 +1,101 @@
 import React from 'react'
-import { Heading, Select, Box } from '@chakra-ui/react'
+import { Heading, Select, Box, SimpleGrid } from '@chakra-ui/react'
 import { ModifiedCharacter } from '../../api/starWars-types'
-import useModifyCharacters, { SORT_FACTOR, SortFactor } from '../../hooks/useModifyCharacters'
+import useModifyCharacters, {
+  SORT_FACTOR,
+  FILTER_FACTOR,
+  SortFactor,
+  FilterFactor,
+} from '../../hooks/useModifyCharacters'
 import CharacterList from '../CharacterList/CharacterList'
+import Selector from '../UI/Selector/Selector'
 
 type AllSpeciesProps = {
   characters: ModifiedCharacter[]
 }
 
+const filterTargets = (characters: ModifiedCharacter[], filterFactor: FilterFactor) => {
+  const filterTargetsObject = characters.reduce(
+    (acc: { [key: string]: string }, character: ModifiedCharacter) => {
+      const value = character[filterFactor]
+      if (!value || typeof value === 'undefined') {
+        return acc
+      } else if (!acc[value]) {
+        acc[value] = value
+      }
```

Comment:
You may start a new if statement instead of using `else if` statement because once the first if statement meets the condition, it returns and stops the flow.